evalml/pipelines/components/transformers/preprocessing/featuretools.py

```diff
@@ -42,6 +45,24 @@ def _make_entity_set(self, X):
             es = ft_es.add_dataframe(dataframe=X, dataframe_name="X", index=self.index)
         return es
 
+    def _filter_features(self, X):
+        features_to_use = []
+        for feature in self.features:
+            input_cols = [f.column_name for f in feature.base_features]
+            if (
+                isinstance(feature, IdentityFeature)
+                and feature.column_name not in X.columns
+            ):
+                continue
+            if not set(input_cols).issubset(set(X.columns)):
```

Comment:
Minor point, but you are converting `X.columns` to a set in multiple places in this loop each pass through. You could do the set conversion once outside of the `for` loop, which should make this faster.